core/node/rules/rule_builder_ae.go

```diff
@@ -101,6 +102,33 @@ func (re *ruleBuilderAEImpl) fail(err error) ruleBuilderAE {
 	return re
 }
 
+func (re *ruleBuilderAEImpl) requireChannelMessagePermissions(params *aeParams) ruleBuilderAE {
+	if params.parsedEvent.Event.Tags != nil {
+		interactionType := params.parsedEvent.Event.Tags.MessageInteractionType
+
+		switch interactionType {
+		case MessageInteractionType_MESSAGE_INTERACTION_TYPE_REACTION,
+			MessageInteractionType_MESSAGE_INTERACTION_TYPE_REDACTION:
+			// reactions - should be allowed if write or react
+			// redactions - should be allowed if write or react (users can always redact their own reactions)
+			return re.requireOneOfChainAuths(
+				params.channelEntitlements(auth.PermissionReact),
+				params.channelEntitlements(auth.PermissionWrite),
```

Comment:
1. can you flip these? I expect that write will be way more common than react and we should check that one first (assuming they're not checked in parallel). Also it keeps it the same as it was before.
2. REPLY should be up here right? Isn't "React/Reply" the setting in the ui?
3. TIP and TRADE should definitely be up here
4. EDIT needs to be up here if REPLY is up here